File: Study Plan/LeetCode 75/0151 Reverse Words in a String/ReverseWordsInAString.cs
/*  
    Problem
    --------------------------------------------------------------------------------
    151. Reverse Words in a String
*/

/*  
    Solution
    --------------------------------------------------------------------------------
    - Runtime 53 ms (Beats 98.84% of users with C#)
    - Memory 40.24 MB (Beats 34.07% of users with C#)
*/

public class Solution
{
    public string ReverseWords(string s)
    {
        var words = s
            .Split(" ", StringSplitOptions.RemoveEmptyEntries)
            .Reverse();
    
        return string.Join(" ", words);  
    }
}

File: Study Plan/LeetCode 75/0151 Reverse Words in a String/ReverseWordsInAString.java
/*  
    Problem
    --------------------------------------------------------------------------------
    151. Reverse Words in a String
*/

/*  
    Solution
    --------------------------------------------------------------------------------
    - Runtime 8 ms (Beats 57.55% of users with Java)
    - Memory 43.00 MB (Beats 67.52% of users with Java)
*/

class Solution {
    public String reverseWords(String s) {
        List<String> words = Arrays.asList(s.split("\\s+"));
        Collections.reverse(words);
        return String.join(" ", words).trim();
    }
}